datafusion/functions/src/core/map.rs

```diff
@@ -75,33 +75,31 @@ fn make_map(args: &[ColumnarValue]) -> Result<ColumnarValue> {
     make_map_batch_internal(key, value, can_evaluate_to_const)
 }
 
+fn get_scalar_from_col(c: &ColumnarValue) -> ScalarValue {
```

Comment:
TODO: Make it a more general helpful function in `datafusion/expr/src/columnar_value.rs`

Reply:
Another TODO is create dataframe API with `map(keys: Vec<Expr>, values: Vec<Expr>)`